src/data/restore.ts

```diff
@@ -254,9 +255,11 @@ export const restoreAppState = (
 
   return {
     ...nextAppState,
-    elementType: AllowedExcalidrawElementTypes[nextAppState.elementType]
-      ? nextAppState.elementType
-      : "selection",
+    elementType:
+      AllowedExcalidrawElementTypes[nextAppState.elementType] &&
+      !isEraserActive(nextAppState)
```

Comment:
if we don't want to persist `eraser` tool across sessions, then let's set it as `false` in `AllowedExcalidrawElementTypes` as that's what it's for.